### DISCUSSION TITLE
how do I stop the map from rotating

### DISCUSSION THREAD

**Discussion by bunk3m** [osmandapp/OsmAnd#16702] [Q&A]
When in driving mode and stopped at a stop light, the map starts rotating and spinning until I start driving again.

What is the setting that I have to enable to stop this rotating behaviour?

Thanks.

**Comment by barteis** [comment]
It could be you have the map in Movement direction rather than Compass direction.  settings / <profile> / general settings / map orientation.

**Reply by bunk3m** [reply]
Thank you @barteis for the suggestion.  

I checked and I do have "Movement" direction set.  And that is actually how I want the map to be oriented when I'm driving.  I like "Up" on the phone to be ahead to where I'm going and I think that is the "Movement" direction.  

If I set it to Compass then the driving direction can, at times, be the narrow side of the phone leaving very little space to see where you're going.

What I don't understand is why when I stop driving that the map rotates around.  It shouldn't be changing the orientation when I'm stopped for 1-2 min waiting for the stop lights to turn green.

**Reply by barteis** [reply]
Interesting.  I would have thought that the Movement and Compass direction would be nearly identical, except when you've stopped.  Then the compass would continue to orient the map to what's in front of you while the lack of movement apparently doesn't.

**Reply by bunk3m** [reply]
@barteis I agree.  That would make sense.

**Comment by dmpr0** [comment]
Hi! 
Do you have the issues on Android Auto or on regular device? We have similar issues on Android Auto. 
Current beta and nightly builds have experimental option **Approximate bearing**. To use it you need to enable **OsmAnd Development** plugin first, then open **Menu** – **Settings** – **Profile** – **Navigation settings** – **Map during navigation** – **Approximate bearing**.

**Reply by bunk3m** [reply]
I am having the issue on regular a regular Android phone/device.   

I can have a look at the "Approximate bearing" option in the next few days. Wouldn't it be easier if the Driving profile Movement direction updating code just stop making adjustments when you aren't moving?
